docs/user/dk_bench.md

```diff
@@ -0,0 +1,185 @@
+```
+
+### Rubric
+
+Each response given by a model is compared to the reference answer and graded on the following 1 - 5 scale by the judge model:
+
+| Score | Criteria |
+| ---------------- | -------- |
+| 1 | The response is entirely incorrect, irrelevant, or does not align with the reference in any meaningful way |
+| 2 | The response partially matches the reference but contains major errors, significant omissions, or irrelevant information |
+| 3 | The response aligns with the reference overall but lacks sufficient detail, clarity, or contains minor inaccuracies |
+| 4 | The response is mostly accurate, aligns closely with the reference, and contains only minor issues or omissions |
+| 5 | The response is fully accurate, completely aligns with the reference, and is clear, thorough, and detailed |
+
+### Output
+
+After running DK-Bench, users will see a report with the question by question score, the average, and the total for their run.
+
+```shell
+# DK-BENCH REPORT
+
+## MODEL: granite-7b-lab
+
+Question #1:     5/5
+Question #2:     5/5
+Question #3:     5/5
+Question #4:     5/5
+Question #5:     2/5
+Question #6:     3/5
+Question #7:     2/5
+Question #8:     3/5
+Question #9:     5/5
+Question #10:     5/5
+----------------------------
+Average Score:   4.00/5
+Total Score:     40/50
+
+Responses and scores written to:
+~/.local/share/instructlab/internal/eval_data/dk_bench/granite-7b-lab/results_2025-02-04T16:18:02.337010.jsonl
+```
+
+The original question, the responses a model gave for that specific question, the reference answer, the score of the judge model are all output to an output file.
+
+The output file could be `.jsonl` (the default and seen below), `.csv`, and `.xlsx`. This can be set using the `--output-file-formats` flag or in the `dk_bench` section of the configuration file.
+
+```json
+{"user_input":"What is the capital of Canada?","response":"The capital of Canada is Ottawa.","reference":"The capital of Canada is Ottawa.","scores":5}
+{"user_input":"What is the capital of Mexico?","response":"The capital of Mexico is Mexico City.","reference":"The capital of Mexico is Mexico City.","scores":5}
+```
+
+## Examples
+
+> [!NOTE]
+> The DK-Bench evaluation requires the environment variables `OPENAI_API_KEY` to be set. It will not run if this variable is not set.
+> To set your OpenAI API Key in the shell run:
+> `export OPENAI_API_KEY=<API KEY HERE>`
+
+DK-Bench has many flags that are applied when the `ilab model evaluate` command to run DK-Bench.
+
+Here is the most basic example of using DK-Bench, by passing in `dk_bench` to the `--benchmark` flag of `ilab model evaluate`, `--input-questions` with the path to the .jsonl dataset, and the path to a local model via `--model`.
+
+```shell
+ilab model evaluate --benchmark dk_bench --input-questions questions.jsonl --model ~/.cache/instructlab/models/instructlab/granite-7b-lab
```

Comment:
silly question, 
can the `--input-questions` be a path to the .jsonl as well?

Reply:
Yes! This could have also been `--input-questions /home/user/path/to/questions.jsonl`

Reply:
was just curious, ty!